src/query/ts/m3db/encoded_unconsolidated_step_iterator.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3db
+
+import (
+	"sync"
+	"time"
+
+	"github.com/m3db/m3/src/dbnode/encoding"
+	"github.com/m3db/m3/src/query/block"
+	xts "github.com/m3db/m3/src/query/ts"
+)
+
+type encodedStepIterUnconsolidated struct {
+	mu          sync.RWMutex
+	lastBlock   bool
+	exhausted   bool
+	started     bool
+	meta        block.Metadata
+	validIters  []bool
+	seriesMeta  []block.SeriesMeta
+	seriesIters []encoding.SeriesIterator
+}
+
+func (b *encodedBlock) stepIterUnconsolidated() block.UnconsolidatedStepIter {
+	return &encodedStepIterUnconsolidated{
+		meta:        b.meta,
+		seriesMeta:  b.seriesMetas,
+		seriesIters: b.seriesBlockIterators,
+		lastBlock:   b.lastBlock,
+	}
+}
+
+type encodedUnconsolidatedStep struct {
```

Comment:
Do we need a new implementation of this? Could we not just share the other implementation (which is identical afaict)?

Reply:
Refactored to use existing